src/app/crash/session.cpp

```diff
@@ -147,18 +149,40 @@ bool Session::isOldSession()
     return true;
 
   int lifespanDays = m_config->keepEditedSpriteDataFor;
-  base::Time sessionTime = base::get_modification_time(verfile);
+  base::Time sessionTime;
+  
+  // Get the session time from the name if possible, to avoid re-scanning when transferring files
+  std::vector<std::string> parts;
+  base::split_string(base::get_file_title(m_path), parts, "-");
+
+  if (parts.size() == 3 && parts[0].size() == 8) {
+    try {
+      sessionTime = base::Time(filenamePartToInt(parts[0].substr(0, 4)),
+                               filenamePartToInt(parts[0].substr(4, 2)),
+                               filenamePartToInt(parts[0].substr(6, 2)),
+                               0,
+                               0,
+                               0);
+    }
+    catch (std::exception& ex) {
```

Comment:
Super minor thing (I think there are a lot of `catch()` statements without `const` that should be `const` out there, so nothing to worry about it):
```suggestion
    catch (const std::exception& ex) {
```